src/mito2/src/cache/write_cache.rs

```diff
@@ -175,19 +176,27 @@ impl WriteCache {
             }])
             .start_timer();
 
+        let cached_value = self
+            .file_cache
+            .local_store()
+            .stat(&cache_path)
+            .await
+            .context(error::OpenDalSnafu)?;
         let reader = self
             .file_cache
             .local_store()
             .reader(&cache_path)
             .await
-            .context(error::OpenDalSnafu)?;
+            .context(error::OpenDalSnafu)?
+            .into_futures_async_read(0..cached_value.content_length());
 
         let mut writer = remote_store
             .writer_with(upload_path)
-            .buffer(DEFAULT_WRITE_BUFFER_SIZE.as_bytes() as usize)
+            .chunk(DEFAULT_WRITE_BUFFER_SIZE.as_bytes() as usize)
             .concurrent(DEFAULT_WRITE_CONCURRENCY)
             .await
-            .context(error::OpenDalSnafu)?;
+            .context(error::OpenDalSnafu)?
+            .into_futures_async_write();
 
         let bytes_written =
             futures::io::copy(reader, &mut writer)
```

Comment:
Yep!

Reply:
@Xuanwo is it available in 0.46? If so, I'd appreciate it if you can send a patch to this PR >_<

Reply:
> is it available in 0.46?

`into_bytes_stream` & `into_bytes_sink` is available in v0.46. `into_stream` & `into_sink` will be available in v0.47.

> If so, I'd appreciate it if you can send a patch to this PR >_<

I'm willing to start a new PR instead of a patch to this PR. This PR is already too large.

Reply:
Make sense.